pkg/adminconfig/component.md

```diff
@@ -0,0 +1,112 @@
+Component for Config Policy Decisions
+==========================
+
+## Summary
+
+Design a component that will evaluate config policy decisions based on the fybrikapplication spec and infrastructure (e.g. deployed clusters).
+
+## Goals
+
+1. Design an input for the policy evaluator
+2. Design an output returned by the policy evaluator.
+3. Design an interface for policy evaluation. A default implementation will not evaluate OPA policies but rather capture hard-coded deployment decisions.
+
+## Proposal
+
+### Input (`adminconfig.EvaluatorInput`)
+
+A dynamic input constructed per a FybrikApplication, per a single dataset.
+It provides general application data such as workload cluster and application properties, as well as dataset details (user requirements, metadata, required actions.
+```
+// WorkloadInfo holds workload details such as cluster/region, type, etc.
+type WorkloadInfo struct {
+	// Cluster where the user workload is running
+	Cluster multicluster.Cluster
+}
+
+// Request is a request to use a specific asset
+type Request struct {
+	// asset identifier
+	DatasetID string
+	// requested interface
+	Interface api.InterfaceDetails
+	// requested usage, e.g. "read": true, "write": false
+	Usage map[api.DataFlow]bool
+}
+
+// EvaluatorInput is an input to Configuration Policies Evaluator.
+// Used to evaluate configuration policies.
+type EvaluatorInput struct {
```

Comment:
If I understand correctly, this is not part of the input interface of the component.  Right?  Isn't it internal, i.e. the structure sent to OPA?  Maybe add a section to this document with internal design.  This would go there, as would a discussion about using OPA as a library and not as an external service.

Reply:
OK. I've split into two interfaces: evaluator <-> manager and evaluator <-> OPA